src/Illuminate/Support/ValidatedInput.php

```diff
@@ -0,0 +1,207 @@
+    }
+
+    /**
+     * Get a subset containing the provided keys with values from the input data.
+     *
+     * @param  array|mixed  $keys
+     * @return array
+     */
+    public function only($keys)
+    {
+        $results = [];
+
+        $input = $this->input;
+
+        $placeholder = new stdClass;
+
+        foreach (is_array($keys) ? $keys : func_get_args() as $key) {
+            $value = data_get($input, $key, $placeholder);
+
+            if ($value !== $placeholder) {
+                Arr::set($results, $key, $value);
+            }
+        }
+
+        return $results;
+    }
+
+    /**
+     * Get all of the input except for a specified array of items.
+     *
+     * @param  array|mixed  $keys
+     * @return array
+     */
+    public function except($keys)
+    {
+        $keys = is_array($keys) ? $keys : func_get_args();
+
+        $results = $this->input;
+
+        Arr::forget($results, $keys);
+
+        return $results;
+    }
+
+    /**
+     * Get the input as a collection.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function collect()
+    {
+        return new Collection($this->input);
+    }
+
+    /**
+     * Get the raw, underlying input array.
+     *
+     * @return array
+     */
+    public function all()
+    {
+        return $this->input;
+    }
+
+    /**
+     * Get the instance as an array.
+     *
+     * @return array
+     */
+    public function toArray()
+    {
+        return $this->all();
+    }
+
+    /**
+     * Dynamically access input data.
+     *
+     * @param  string  $name
+     * @return mixed
+     */
+    public function __get($name)
+    {
+        return $this->input[$name];
+    }
+
+    /**
+     * Dynamically set input data.
+     *
+     * @param  string  $name
+     * @param  mixed  $value
+     * @return mixed
+     */
+    public function __set($name, $value)
+    {
+        $this->input[$name] = $value;
+    }
+
+    /**
+     * Determine if an input key is set.
+     *
+     * @return bool
+     */
+    public function __isset($name)
+    {
+        return isset($this->input[$name]);
+    }
+
+    /**
+     * Remove an input key.
+     *
+     * @param  string  $name
+     * @return void
+     */
+    public function __unset($name)
+    {
+        unset($this->input[$name]);
+    }
+
+    /**
+     * Determine if an item exists at an offset.
+     *
+     * @param  mixed  $key
+     * @return bool
+     */
+    #[\ReturnTypeWillChange]
+    public function offsetExists($key)
+    {
+        return isset($this->input[$key]);
+    }
+
+    /**
+     * Get an item at a given offset.
+     *
+     * @param  mixed  $key
+     * @return mixed
+     */
+    #[\ReturnTypeWillChange]
+    public function offsetGet($key)
+    {
+        return $this->input[$key];
+    }
+
+    /**
+     * Set the item at a given offset.
+     *
+     * @param  mixed  $key
+     * @param  mixed  $value
+     * @return void
+     */
+    #[\ReturnTypeWillChange]
```

Comment:
Why do we need this in a completely new class? The proper return typehint can be added and the annotation can be removed.

Reply:
This is for PHP 8.1 compatibility. We've been rolling these out everywhere since a month now.

Reply:
Yeah, I know. It's one thing to not add the return typehint (and add the annotation instead) on existing classes to avoid a BC break, but there's no reason to add this annotation on a completely new class to which a return typehint can be added from the start so that there's no need for a breaking change in the future.

Reply:
@X-Coder264 there's no mixed return type pre-PHP 8.0

Reply:
I know. But there is a void and bool return typehint for `offsetExists`, `offsetSet`, and `offsetUnset`. That's why I said above "most of the methods", not all.

Reply:
Ah. Tbvh I think it's better to keep consistency for now and do all of these changes in a later release.